cpp/src/io/orc/reader_impl.cu

```diff
@@ -434,59 +457,114 @@ class aggregate_orc_metadata {
     return selected_stripes_mapping;
   }
 
+  /**
+   * @brief Adds column as per the request and saves metadata about children.
+   *        Struct children are in the same level as struct, only list column
+   *        children are pushed to next level.
+   *
+   * @param selection A vector that saves list of columns as per levels of nesting.
+   * @param types A vector of schema types of columns.
+   * @param level current level of nesting.
+   * @param id current column id that needs to be added.
+   * @param has_timestamp_column True if timestamp column present and false otherwise.
+   *
+   * @return returns number of child columns
+   */
+  uint32_t add_column(std::vector<std::vector<orc_column_meta>> &selection,
+                      std::vector<SchemaType> const &types,
+                      const size_t level,
+                      const uint32_t id,
+                      bool &has_timestamp_column,
+                      bool &has_list_column)
```

Comment:
Should we set these separately? Maybe two calls to `std::any` on `types`?
As is, this function does two separate things and it reduces readability.

Reply:
You might have to write same logic that is there for selecting column since you have to iterate over given names and also there child columns, which is almost same logic.